### PR TITLE
authsidecar config missing in houston config has been re-added

### PR DESCRIPTION
**Pull request by pgvishnuram**
## Description

This PR fixes missing config's to enable authsidecar for airflow deployments 

## PR Title

Fix missing configs of authsidecar

## 🎟 Issue(s)

https://github.com/astronomer/issues/issues/3853

### CODE REVIEW

**Review by aliotta — APPROVED** [review]

**Review by danielhoherd — CHANGES_REQUESTED** [review]
Looks good but needs tests.

**Comment by pgvishnuram** [comment]
yes sure - i will need to add entire feature test - previously i have not added them

**Comment by andriisoldatenko** [comment]
LGTM, could you pls fix CI platform-1-21-2 ?

File: tests/test_authsidecar.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+                "charts/alertmanager/templates/alertmanager-statefulset.yaml",
+                "charts/alertmanager/templates/alertmanager-auth-sidecar-configmap.yaml",
+                "charts/alertmanager/templates/alertmanager-service.yaml",
+                "charts/alertmanager/templates/alertmanager-networkpolicy.yaml",
+                "charts/alertmanager/templates/ingress.yaml",
+            ],
+        )
+
+        assert len(docs) == 5
+        doc = docs[0]
```
Comment (danielhoherd) [inline tests/test_authsidecar.py:28]: Why are we rendering 5 templates but only making assertions against 1?

File: tests/test_authsidecar.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        assert doc["kind"] == "StatefulSet"
+        assert doc["apiVersion"] == "apps/v1"
+        assert doc["metadata"]["name"] == "RELEASE-NAME-alertmanager"
+        assert "auth-proxy" == doc["spec"]["template"]["spec"]["containers"][1]["name"]
```
Comment (danielhoherd) [inline tests/test_authsidecar.py:32]:
```suggestion
        assert doc["spec"]["template"]["spec"]["containers"][1]["name"] == "auth-proxy"
```

File: tests/test_authsidecar.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+        assert "Service" == jmespath.search("kind", docs[2])
```
Comment (danielhoherd) [inline tests/test_authsidecar.py:34]:
```suggestion
        assert jmespath.search("kind", docs[2]) == "Service"
```

File: tests/test_authsidecar.py [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        assert "RELEASE-NAME-alertmanager" == jmespath.search("metadata.name", docs[2])
```
Comment (danielhoherd) [inline tests/test_authsidecar.py:35]:
```suggestion
        assert jmespath.search("metadata.name", docs[2]) == "RELEASE-NAME-alertmanager"
```

File: tests/test_authsidecar.py [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+        assert "ClusterIP" == jmespath.search("spec.type", docs[2])
```
Comment (danielhoherd) [inline tests/test_authsidecar.py:36]:
```suggestion
        assert jmespath.search("spec.type", docs[2]) == "ClusterIP"
```

File: tests/test_authsidecar.py [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+                "charts/prometheus/templates/prometheus-statefulset.yaml",
+                "charts/prometheus/templates/prometheus-auth-sidecar-configmap.yaml",
+                "charts/prometheus/templates/prometheus-service.yaml",
+                "charts/prometheus/templates/prometheus-networkpolicy.yaml",
+                "charts/prometheus/templates/ingress.yaml",
+            ],
+        )
+
+        assert len(docs) == 5
+        doc = docs[0]
```
Comment (danielhoherd) [inline tests/test_authsidecar.py:58]: Again, is there a reason we're rendering 5 templates and only asserting against 1? I feel like we should be making assertions about all of them if we're going to render them all.

**Review by danielhoherd — CHANGES_REQUESTED** [review]
Looks generally good. I have a few suggestions on tests, but looking back at them I think they may just be [bike-shedding](https://en.wiktionary.org/wiki/bikeshedding). More importantly is a question about why we are rendering multiple templates but only making assertions against one. It would be great to make assertions against all relevant templates, and remove any templates that are not relevant to the test case.

File: tests/test_authsidecar.py (hunk 6 above)
Comment (pgvishnuram) [reply tests/test_authsidecar.py:58]: I am asserting statefulset and service

Comment (pgvishnuram) [reply tests/test_authsidecar.py:58]: I still need to add more test cases to test against all the mentioned templates

**Comment by vishwas-astro** [comment]
>  [bike-shedding](https://en.wiktionary.org/wiki/bikeshedding). 
so true in a lot of code reviews I have seen and possibly done myself :)

**Review by danielhoherd — APPROVED** [review]

**Pull request merged**